src/models/oracle/facade.js

```diff
@@ -24,112 +24,129 @@
+                partyIdentifier: params.ID,
+                currency: query.currency || payload.currency
+              })
+              break
+            }
           }
+        } else {
+          url = Mustache.render(oracleEndpointModel[0].value + Enums.EndPoints.FspEndpointTemplates.ORACLE_PARTICIPANTS_TYPE_ID_CURRENCY, {
+            partyIdType: type,
+            partyIdentifier: params.ID,
+            currency: query.currency || payload.currency
+          })
         }
       } else {
-        url = Mustache.render(oracleEndpointModel[0].value + Enums.endpoints.oracleParticipantsTypeIdCurrency, {
-          partyIdType: type,
-          partyIdentifier: req.params.ID,
-          currency: req.query.currency || req.payload.currency
-        })
+        Logger.error(`Oracle type:${type} and currency:${query.currency || payload.currency} not found`)
+        throw ErrorHandler.Factory.createFSPIOPError(ErrorHandler.Enums.FSPIOPErrorCodes.ADD_PARTY_INFO_ERROR, `Oracle type:${type} and currency:${query.currency || payload.currency} not found`).toApiErrorObject()
       }
     } else {
-      Logger.error(`Oracle type:${type} and currency:${req.query.currency || req.payload.currency} not found`)
-      return null
-    }
-  } else {
-    oracleEndpointModel = await oracleEndpoint.getOracleEndpointByType(type)
-    if (oracleEndpointModel.length > 0) {
-      if (oracleEndpointModel.length > 1) {
-        for (const record in oracleEndpointModel) {
-          if (oracleEndpointModel.hasOwnProperty(record) && record.isDefault) {
-            url = Mustache.render(record.value + Enums.endpoints.oracleParticipantsTypeId, {
-              partyIdType: type,
-              partyIdentifier: req.params.ID
-            })
-            break
+      oracleEndpointModel = await oracleEndpoint.getOracleEndpointByType(type)
+      if (oracleEndpointModel.length > 0) {
+        if (oracleEndpointModel.length > 1) {
+          for (const record of oracleEndpointModel) {
+            if (record.isDefault) {
+              url = Mustache.render(record.value + Enums.EndPoints.FspEndpointTemplates.ORACLE_PARTICIPANTS_TYPE_ID, {
+                partyIdType: type,
+                partyIdentifier: params.ID
+              })
+              break
+            }
           }
+        } else {
+          url = Mustache.render(oracleEndpointModel[0].value + Enums.EndPoints.FspEndpointTemplates.ORACLE_PARTICIPANTS_TYPE_ID, {
+            partyIdType: type,
+            partyIdentifier: params.ID
+          })
         }
       } else {
-        url = Mustache.render(oracleEndpointModel[0].value + Enums.endpoints.oracleParticipantsTypeId, {
-          partyIdType: type,
-          partyIdentifier: req.params.ID
-        })
+        Logger.error(`Oracle type:${type} not found`)
+        throw ErrorHandler.Factory.createFSPIOPError(ErrorHandler.Enums.FSPIOPErrorCodes.ADD_PARTY_INFO_ERROR, `Oracle type:${type} not found`).toApiErrorObject()
```

Comment:
`throw ErrorHandler.Factory.createFSPIOPError(ErrorHandler.Enums.FSPIOPErrorCodes.ADD_PARTY_INFO_ERROR, `Oracle type:${type} not found`).toApiErrorObject()` --> `throw ErrorHandler.Factory.createFSPIOPError(ErrorHandler.Enums.FSPIOPErrorCodes.ADD_PARTY_INFO_ERROR, `Oracle type:${type} not found`)`